crates/uv-auth/src/middleware.rs

```diff
@@ -374,6 +397,9 @@ impl AuthMiddleware {
             .as_ref()
             .is_ok_and(|response| response.error_for_status_ref().is_ok())
         {
+            if let (Some(index_url), Some(keyring)) = (index_url, &self.keyring) {
```

Comment:
This is currently only storing credentials on successful authentication if this is an index URL.